Unable to resolve "event-target-shim/index" from "node_modules/@livekit/react-native-webrtc/src/MediaDevices.ts"

**Describe the bug**
After installing all dependencies as per document we are facing "Unable to resolve "event-target-shim/index" from "node_modules/@livekit/react-native-webrtc/src/MediaDevices.ts" error.

**To Reproduce**

Steps to reproduce the behavior:
1. Install all dependencies as per document.
2. Run npx expo run:ios   
3. See error

**Expected behavior**

App should work without any issues

**Screenshots**

<img width="822" alt="Image" src="https://github.com/user-attachments/assets/b7c87c9d-89a0-48a8-8976-52662086bb1f" />


**Device Info:**
 - Device:  Iphone 16
 - OS: 18

**Dependencies Info (please reference your package-lock.json or yarn.lock file, not just your package.json):**

 - @livekit/react-native: 
 - livekit-client: 
 - react-native-webrtc: 

**Additional context**
We are having bare managed flow for react native expo.

Above issue resolved after adding following in metro config, but facing new error after this.

```
config.resolver.resolveRequest = (context, moduleName, platform) => {
  if (
    // If the bundle is resolving "event-target-shim" from a module that is part of "react-native-webrtc".
    // moduleName.startsWith('event-target-shim') &&¸
    context.originModulePath.includes('react-native-webrtc')
  ) {
    // Resolve event-target-shim relative to the react-native-webrtc package to use v6.
    // React Native requires v5 which is not compatible with react-native-webrtc.
    const eventTargetShimPath = resolveFrom(
      context.originModulePath,
      moduleName,
    );

    return {
      filePath: eventTargetShimPath,
      type: 'sourceFile',
    };
  }

  // Ensure you call the default resolver.
  return context.resolveRequest(context, moduleName, platform);
};

```

 ERROR  Error: Cannot find module './EventEmitter'
Require stack:
- /Users/xyz/Documents/xxxx/node_modules/@livekit/react-native-webrtc/src/index.ts/noop.js

--- a/package.json
+++ b/package.json
@@ -57,7 +57,6 @@
     "@livekit/mutex": "^1.1.1",
     "array.prototype.at": "^1.1.1",
     "base64-js": "1.5.1",
-    "event-target-shim": "6.0.2",
     "events": "^3.3.0",
     "loglevel": "^1.8.0",
     "promise.allsettled": "^1.0.5",

--- a/src/audio/MediaRecorder.ts
+++ b/src/audio/MediaRecorder.ts
@@ -1,10 +1,11 @@
 import type { MediaStream } from '@livekit/react-native-webrtc';
-import { addListener } from '../events/EventEmitter';
 import {
   EventTarget,
   Event,
-  defineEventAttribute,
-} from 'event-target-shim/index';
+  getEventAttributeValue,
+  setEventAttributeValue,
+} from '@livekit/react-native-webrtc';
+import { addListener } from '../events/EventEmitter';
 import { toByteArray } from 'base64-js';
 import LiveKitModule from '../LKNativeModule';
 import { log } from '../logger';
@@ -123,6 +124,54 @@ export class MediaRecorder extends EventTarget<MediaRecorderEventMap> {
       new BlobEvent('dataavailable', { data: { byteArray: data } })
     );
   }
+
+  get ondataavailable() {
+    return getEventAttributeValue(this, 'dataavailable');
+  }
+
+  set ondataavailable(value) {
+    setEventAttributeValue(this, 'dataavailable', value);
+  }
+
+  get onerror() {
+    return getEventAttributeValue(this, 'error');
+  }
+
+  set onerror(value) {
+    setEventAttributeValue(this, 'error', value);
+  }
+
+  get onpause() {
+    return getEventAttributeValue(this, 'pause');
+  }
+
+  set onpause(value) {
+    setEventAttributeValue(this, 'pause', value);
+  }
+
+  get onresume() {
+    return getEventAttributeValue(this, 'resume');
+  }
+
+  set onresume(value) {
+    setEventAttributeValue(this, 'resume', value);
+  }
+
+  get onstart() {
+    return getEventAttributeValue(this, 'start');
+  }
+
+  set onstart(value) {
+    setEventAttributeValue(this, 'start', value);
+  }
+
+  get onstop() {
+    return getEventAttributeValue(this, 'stop');
+  }
+
+  set onstop(value) {
+    setEventAttributeValue(this, 'stop', value);
+  }
 }
 
 /**
@@ -144,15 +193,3 @@ class BlobEvent<TEventType extends string> extends Event<TEventType> {
     this.data = eventInitDict.data;
   }
 }
-
-/**
- * Define the `onxxx` event handlers.
- */
-const proto = MediaRecorder.prototype;
-
-defineEventAttribute(proto, 'dataavailable');
-defineEventAttribute(proto, 'error');
-defineEventAttribute(proto, 'pause');
-defineEventAttribute(proto, 'resume');
-defineEventAttribute(proto, 'start');
-defineEventAttribute(proto, 'stop');

--- a/src/index.tsx
+++ b/src/index.tsx
@@ -161,6 +161,7 @@ export * from './useParticipant'; // deprecated
 export * from './useRoom'; // deprecated
 export * from './logger';
 export * from './audio/AudioManager';
+export * from './audio/MediaRecorder';
 
 export {
   AudioSession,

